source/tests/threadingzeug-test/parallel_for_test.cpp

```diff
@@ -7,31 +7,186 @@
 
 using namespace threadingzeug;
 
-class parallel_for_test : public testing::Test
+class parallelFor_test : public testing::Test
 {
 public:
-    parallel_for_test()
+    parallelFor_test()
     {
     }
 
 protected:
 };
 
-TEST_F(parallel_for_test, IncrementVector)
+TEST_F(parallelFor_test, ParellelIncrementVector)
 {
-    const auto start = 3, end = 27;
+    const auto start = 3u;
+    const auto end = 27u;
     auto vec = std::vector<int>(27, 0);
 
-    parallel_for(start, end, [&vec] (int i)
-        {
-            std::this_thread::sleep_for(std::chrono::nanoseconds{1000000});
-            vec[i] = 1;
-        });
+    parallelFor(start, end, [&vec] (int i) {
+        std::this_thread::sleep_for(std::chrono::nanoseconds{1000000});
+        vec[i] = 1;
+    });
 
-    auto allTrue = true;
+    for (auto i = start; i < end; ++i)
+    {
+        ASSERT_EQ(vec[i], 1);
+    }
+}
+
+TEST_F(parallelFor_test, SequentialIncrementVector)
+{
+    const auto start = 3u;
+    const auto end = 27u;
+    auto vec = std::vector<int>(27, 0);
+
+    sequentialFor(start, end, [&vec] (int i) {
+        std::this_thread::sleep_for(std::chrono::nanoseconds{1000000});
+        vec[i] = 1;
+    });
+
+    for (auto i = start; i < end; ++i)
+    {
+        ASSERT_EQ(vec[i], 1);
+    }
+}
+
+TEST_F(parallelFor_test, ParellelSequentialEquality)
+{
+    const auto start = 3u;
+    const auto end = 27u;
+    auto vec1 = std::vector<int>(27, 0);
+    auto vec2 = std::vector<int>(27, 0);
+
+    parallelFor(start, end, [&vec1] (int i) {
+        std::this_thread::sleep_for(std::chrono::nanoseconds{1000000});
+        vec1[i] = 1;
+    });
+
+    sequentialFor(start, end, [&vec2] (int i) {
+        std::this_thread::sleep_for(std::chrono::nanoseconds{1000000});
+        vec2[i] = 1;
+    });
 
     for (auto i = start; i < end; ++i)
-        allTrue &= (vec[i] == 1);
+    {
+        ASSERT_EQ(vec1[i], vec2[i]);
+    }
+}
+
+TEST_F(parallelFor_test, SizeTCompilation)
+{
+    size_t start = 0;
+    size_t end = 1;
+
+    parallelFor(start, end, [] (int i) {
```

Comment:
http://stackoverflow.com/questions/11603818/why-is-there-ambiguity-between-uint32-t-and-uint64-t-when-using-size-t-on-mac-os